src/modules/taskList.js

```diff
@@ -6,15 +6,47 @@ export default class {
     this.list = list;
   }
 
+  regenerateValues(datalist) {
+    const takslist = [];
+    datalist.forEach((entry) => {
+      takslist.push(new Task(
+        entry.value,
+        entry.index,
+        entry.completed,
+      ));
+    });
+
+    this.list = takslist;
+  }
+
+  exportListValues() {
+    const datalist = [];
+    this.list.forEach((task) => {
+      datalist.push({
+        value: task.value,
+        index: task.index,
+        completed: task.completed,
+      });
+    });
+
+    return datalist;
+  }
+
   add(value) {
-    this.list.push(new Task(value, this.list.length));
+    const task = new Task(value, this.list.length + 1);
+    this.list.push(task);
+    return task;
   }
 
   remove(index) {
     this.list.splice(index, 1);
     this.#updateTasksIndex();
   }
```

Comment:
- [ ] Great work on implementing the index issues :blush: but you still have an index problem in your delete method because when we try to delete the first todo then the second todo is deleted if we try to delete the second todo and the third todo is deleted. Please fix this issue :+1:

Reply:
Oh, that's true! I think the three functions (add, remove, edit) should be working properly now